libs/seiscomp/processing/magnitudes/MLc.cpp

```diff
@@ -264,12 +264,20 @@ MagnitudeProcessor::Status MagnitudeProcessor_MLc::computeMagnitude(
 	}
 
 	SEISCOMP_DEBUG("  + distance mode: %s", distanceMode.c_str());
+
+	double hDistanceKm = Math::Geo::deg2km(delta);
+	double vDistanceKm = 0;
+	if ( !receiver && distanceMode == "hypocentral" ) {
+		return MetaDataRequired;
+	}
+	if ( receiver ) {
+		vDistanceKm = abs(receiver->elevation() / 1000. + depth);
```

Comment:
maybe abs() is not required here and it even would hide the information if the event is below or  above the sensor  which is also reported in debug output.

Reply:
Good point.